### ISSUE TITLE
Make handleAction defaultState optional or move it to a separate function

### ISSUE THREAD

**Issue by mxl** [reduxactions/redux-actions#254]
In my opinion handling `defaultState` in `handleAction` function does not belong to core features of this module and `defaultState` parameter should be optional or its handling should be moved to separate function (`handleActionWithDefaultState`). Requiring it makes hard to decouple reducers definition from initial state definition.

Of course, in JavaScript I can simply pass `{}` as default state (which is actually wrong if initial state must have some defined properties), but [in TypeScript](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/db4cffca9198844e6071865536cae9ee7eddccf7/types/redux-actions/index.d.ts#L128) I have to pass full initial state object which is not convenient because:
1. I want reducers definition to be separate from initial state;
2. I do not want to create reducers builder function like `function createReducer(initialState: MyState): Reducer<MyState>`;
3. the right way to pass initial state is to pass it as second argument to redux `createStore()` function; passing it in helper methods like `handleAction` is a "workaround" solution to a situation when state is undefined.

To sum this up: passing `defaultState` is like adding a sugar to a tea, some of us like it (maybe even most of us), but it's not required.

**Comment by mslipper** [comment]
Doesn't delegating responsibility for the reducer's initial state to the `createStore` method make creating portable reducers harder? Rather than having the reducer define its own initial state, you're relying on the invocation of `createStore` to do it.

To your second point - mind helping me understand your aversion to creating reducer builder functions?

To your third point - is defining an initial state in your reducer really an antipattern? The Redux book seems to suggest that as a best practice when defining reducers.

**Comment by mslipper** [comment]
Closing this since it's super old now - please reopen if you'd like to discuss more.
